jgrapht-core/src/main/java/org/jgrapht/alg/intervalgraph/LexBreathFirstSearch.java

```diff
@@ -0,0 +1,217 @@
+            
+            // set index of v to be n + 1 - i
+            ordering.add(currentVertex);
+            
+            // for every unvisited neighbor w of v, append i to label of w
+            for(V neighbor : Graphs.neighborListOf(graph, currentVertex)) {
+                String oldLabel = vertexToLabelMap.get(neighbor);
+                String newLabel = oldLabel + Integer.toString(i);
+                
+                // remove neighbor from previous list
+                ArrayList<V> oldList = labelToVertexMap.get(oldLabel);
+                oldList.remove(neighbor);
+                
+                // if there are no vertices left in this list, delete the entry
+                if(oldList.isEmpty()) {
+                    labelToVertexMap.remove(oldLabel);
+                } else {
+                    // else we replace the list with its updated version 
+                    labelToVertexMap.replace(oldLabel, oldList);
+                }
+                
+                // now add neighbor to the list of the new label
+                if(labelToVertexMap.containsKey(newLabel)) {
+                    ArrayList<V> newList = labelToVertexMap.get(newLabel);
+                    newList.add(neighbor);
+                    labelToVertexMap.replace(newLabel, newList);
+                } else {
+                    ArrayList<V> newList = new ArrayList<V>();
+                    newList.add(neighbor);
+                    labelToVertexMap.put(newLabel, newList);
+                }
+                
+                // associate neighbor with its new label
+                vertexToLabelMap.replace(neighbor, newLabel);
+            }
+            
+            // mark current vertex as visited, i.e. remove its entries from the maps
+            vertexToLabelMap.remove(currentVertex);
+            candidates.remove(currentVertex);
+            labelToVertexMap.replace(maxLabel, candidates);
+        }
+    
+        // return ordering -- thanks Java for not being able to create an array from a list with generic types for some reason
+        V[] orderingArray = null;
+        orderingArray = ordering.toArray(orderingArray);
+        
+        return orderingArray;
+    }
+    
+    class LBFSComparator implements Comparator<V>
+    {
+        private final V[] prevOrdering;
+        
+        public LBFSComparator(V[] prevOrdering) {
+            this.prevOrdering = prevOrdering;
+        }
+        
+        // apparantly Java has no indexOf method...
+        private int indexOf(V vertex) throws NoSuchElementException{
+            for(int index = 0; index < prevOrdering.length; index++) {
+                if(prevOrdering[index].equals(vertex)) {
+                    return index;
+                }
+            }
+            throw new NoSuchElementException("Previous ordering is on a different vertex set.");
+        }
+        
+        @Override
+        public int compare(V vertex1, V vertex2)
```

Comment:
This is currently a O(n^2) algorithm which makes sorting O(n^2*n log(n)) = O(n^3 log(n)). Maybe use a map vertex -> position (with a hashmap)